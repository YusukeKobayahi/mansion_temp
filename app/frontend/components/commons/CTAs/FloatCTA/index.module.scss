@import "~/styles/variables";
@import "~/styles/utils";

.container {
  position: fixed;
  bottom: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  padding: 15px 0;
  background-color: rgb(224 239 240 / 80%);
  opacity: 0;
  display: none;

  &.displayStyle {
    display: block;
    animation: fadeIn 0.7s cubic-bezier(0.33, 1, 0.68, 1) 1 forwards;
  }

  @include screenMd {
    padding: 10px;
  }
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.description {
  text-align: center;
  line-height: 1.5;
  margin-bottom: 5px;
  color: #746744;
  font-size: 16px;
  font-weight: bold;

  .spOnly {
    display: none;
  }

  @include screenMd {
    font-size: 14px;

    .pcOnly {
      display: none;
    }

    .spOnly {
      display: inline;
    }
  }
}

.btn {
  width: 500px;
  height: 66px;
  display: block;
  margin: 0 auto;
  padding: 16px 32px 16px 48px;
  border-radius: 60px;
  font-size: 24px;
  font-weight: bold;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #ffffff;
  position: relative;
  background-color: #f15e3a;

  @include screenMd {
    width: 100%;
    font-size: 20px;
    padding-left: 50px;
    max-width: 400px;
    padding: 10px 24px 10px 60px;
    height: 65px;
  }

  span {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #ffffff;
    color: #f15e3a;
    font-size: 18px;
    width: 55px;
    height: 55px;
    line-height: 1.2;
    border-radius: 50px;
    position: absolute;
    top: 50%;
    left: 8px;
    transform: translateY(-50%);

    @include screenMd {
      width: 55px;
      height: 55px;
      line-height: 1.3;
      font-size: 16px;
    }
  }

  &::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 30px;
    width: 10px;
    height: 10px;
    border-top: 2px solid #ffffff;
    border-right: 2px solid #ffffff;
    transform: translateY(-50%) rotate(45deg);

    @include screenMd {
      right: 20px;
    }
  }
}
