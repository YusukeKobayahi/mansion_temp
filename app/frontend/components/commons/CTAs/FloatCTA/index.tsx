import { useEffect, useState } from "react";
import classnames from "classnames";
import styles from "./index.module.scss";
import { clickMembers } from "~/lib/utils";

type FloatCTAProps = {
  page: "msl_ichiran" | "msl_prefectures";
};

const FloatCTA: React.FC<FloatCTAProps> = ({ page }: FloatCTAProps) => {
  const [display, setDisplay] = useState(false);
  useEffect(() => {
    const scrollEvent = () => {
      if (window.pageYOffset > 0) setDisplay(true);
      else setDisplay(false);
    };
    window.addEventListener("scroll", scrollEvent);

    return () => {
      window.removeEventListener("scroll", scrollEvent);
    };
  }, []);
  const clickCTA = () => {
    clickMembers({
      action: "msl_cta_click",
      category: page,
      label: "msl_lp_flooting",
      value: 1,
    });
  };
  return (
    <section
      className={classnames(
        styles.container,
        display ? styles.displayStyle : ""
      )}
    >
      <p className={styles.description}>
        ＼ 複数の不動産会社があなたの理想の物件を提案／
      </p>
      <button className={styles.btn} onClick={clickCTA}>
        <span>
          完全
          <br />
          無料
        </span>
        まずは希望の条件を登録してみる
      </button>
    </section>
  );
};

export default FloatCTA;
