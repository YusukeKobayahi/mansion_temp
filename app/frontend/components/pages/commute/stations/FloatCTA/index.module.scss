@import "~/styles/variables";
@import "~/styles/utils";

.container {
  position: fixed;
  bottom: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  padding: 15px 0;
  background-color: rgb(224 239 240 / 80%);

  @include screenMd {
    padding: 10px;
  }
}

.children {
  margin: 0 auto;
  width: 500px;
  display: flex;
  justify-content: space-between;

  @include screenMd {
    width: 100%;
    padding: 10px;
  }
}
