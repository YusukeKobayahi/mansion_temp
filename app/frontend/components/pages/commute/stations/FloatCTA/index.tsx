import styles from "./index.module.scss";
import classnames from "classnames";

type Visility = "pcOnly" | "spOnly" | "";
type FloatCTAProps = {
  visility?: Visility;
  children: React.ReactNode;
};

const FloatCTA: React.FC<FloatCTAProps> = ({
  children,
  visility = "",
}: FloatCTAProps) => {
  return (
    <section className={classnames(styles.container, visility)}>
      <div className={styles.children}>{children}</div>
    </section>
  );
};

export default FloatCTA;
