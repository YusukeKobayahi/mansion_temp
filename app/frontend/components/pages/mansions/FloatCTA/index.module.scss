@import "~/styles/variables";
@import "~/styles/utils";

.container {
  position: fixed;
  bottom: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  background-color: rgb(224 239 240 / 80%);
  opacity: 0;
  display: none;

  &.displayStyle {
    display: block;
    animation: fadeIn 0.7s cubic-bezier(0.33, 1, 0.68, 1) 1 forwards;
  }
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.inner {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: flex-end;
  padding: 20px 0;

  @include screenMd {
    padding: 10px;
    justify-content: space-between;
  }
}

.btn {
  width: 460px;
  margin: 0 5px;
  box-sizing: border-box;
  border-radius: 40px;
  font-size: 19px;
  cursor: pointer;
  position: relative;
  font-weight: bold;

  @include screenMd {
    margin: 0;
    width: calc((100% - 4px) / 2);
    font-size: 3.2vw;
    text-align: center;
  }

  &.formBtn {
    background-color: #ffffff;
    border: 2px solid #f15e3a;
    color: #f15e3a;
    line-height: 56px;

    @include screenMd {
      line-height: 1.3;
      padding: 10px 10px 10px 0;
    }

    &::after {
      border-top: 2px solid #f15e3a;
      border-right: 2px solid #f15e3a;

      @include screenMd {
        border-top-width: 1px;
        border-right-width: 1px;
      }
    }
  }

  &.lpBtn {
    background-color: #f15e3a;
    color: #ffffff;
    line-height: 60px;

    @include screenMd {
      padding: 12px 12px 12px 0;
      line-height: 1.3;
    }

    &::after {
      border-top: 2px solid #ffffff;
      border-right: 2px solid #ffffff;

      @include screenMd {
        border-top-width: 1px;
        border-right-width: 1px;
      }
    }
  }

  &::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 30px;
    width: 10px;
    height: 10px;
    transform: translateY(-50%) rotate(45deg);

    @include screenMd {
      width: 7px;
      height: 7px;
      right: 10px;
    }
  }
}
