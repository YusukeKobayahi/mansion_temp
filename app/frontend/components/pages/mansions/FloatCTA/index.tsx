import { useEffect, useState } from "react";
import { clickMembers } from "~/lib/utils";
import classnames from "classnames";
import styles from "./index.module.scss";
import { GoogleAnalyticsEvent } from "~/lib/types";

type FloatCTAProps = {
  uniqueCode: string;
};

const FloatCTA: React.FC<FloatCTAProps> = ({ uniqueCode }: FloatCTAProps) => {
  const [display, setDisplay] = useState(false);
  useEffect(() => {
    const scrollEvent = () => {
      if (window.pageYOffset > 0) setDisplay(true);
      else setDisplay(false);
    };
    window.addEventListener("scroll", scrollEvent);

    return () => {
      window.removeEventListener("scroll", scrollEvent);
    };
  }, [uniqueCode]);
  const clickCTA = (label: GoogleAnalyticsEvent["label"]) => {
    clickMembers({
      action: "msl_cta_click",
      category: "msl_shosai",
      label: label,
      value: 1,
    });
  };
  return (
    <section
      className={classnames(
        styles.container,
        display ? styles.displayStyle : ""
      )}
    >
      <div className={styles.inner}>
        <button
          className={classnames(styles.btn, styles.formBtn)}
          onClick={() => clickCTA("msl_flooting_left")}
        >
          この物件の最新情報を
          <br className={"spOnly"} />
          受け取る
        </button>
        <button
          className={classnames(styles.btn, styles.lpBtn)}
          onClick={() => clickCTA("msl_flooting_right")}
        >
          未公開物件の
          <br className={"spOnly"} />
          提案を手に入れる
        </button>
      </div>
    </section>
  );
};

export default FloatCTA;
