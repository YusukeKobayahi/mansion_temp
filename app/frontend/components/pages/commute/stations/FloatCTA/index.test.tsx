import { render } from "@testing-library/react";
import FloatCTA from ".";

describe("components/pages/commute/stations/FloatCTA", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(
      <FloatCTA visility={""}>
        <button>会員登録</button>
      </FloatCTA>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
