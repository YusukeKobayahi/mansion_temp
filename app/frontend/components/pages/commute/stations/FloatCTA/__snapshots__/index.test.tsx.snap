// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/commute/stations/FloatCTA 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <div
      class="children"
    >
      <button>
        会員登録
      </button>
    </div>
  </section>
</DocumentFragment>
`;
